### PR TITLE
Update weapon_bonus.py

### PR DESCRIPTION
**Pull request by CaseyMakes**
Created a dictionary (weapon_skill_bonuses) to store the bonuses for each weapon and corresponding weapon skill. Each weapon is a key in the dictionary, and the value is another dictionary mapping weapon skills to their bonuses. Using dictionaries for lookups is faster and more efficient than multiple if-elif statements. The code is shorter and clearer, making it easier to understand and maintain.

### CODE REVIEW

File: weapon_bonus.py [hunk 1]
```diff
@@ -68,7 +68,8 @@ def get_weapon_bonus(main_wpn_name, rng_wpn_name, ws_name):
         "Drepanum": {"Spiral Hell": 1.0},
         "Xoanon": {"Retribution": 0.2},
         "Khatvanga": {"Shattersoul": 0.1},
-        "Vajra": {"Mandalic Stab": 0.495, "Pyrrhic Kleos": 0.495},
+        "Vajra": {"Mandalic Stab": 0.495},
+        "Terpsichore": {"Pyrrhic Kleos": 0.495},
```
Comment (IzaKastra) [inline weapon_bonus.py:72]: Terpsichore is already included on line 45.

File: weapon_bonus.py [hunk 2]
```diff
@@ -7,245 +7,95 @@
...
+        "Tishtrya": {"Realmrazer": 0.1},
+        "Mjollnir": {"Randgrith": 0.68},
+        "Yagrush": {"Mystic Boon": 0.495},
+        "Idris": {"Exudation": 0.15},
+        "Epeolatry": {"Dimidiation": 0.15},
+        "Kikoku": {"Blade: Metsu": 0.68},
+        "Kannagi": {"Blade: Hi": 0.1},
+        "Nagi": {"Blade: Kamu": 0.495},
+        "Kenkonken": {"Stringing Pummel": 0.495},
+        "Heishi": {"Blade: Shun": 0.1},
```
Comment (IzaKastra) [inline weapon_bonus.py:31]: This should be "Heishi Shorinken" to match the "Name" key in the Heishi dictionary.

File: weapon_bonus.py [hunk 3]
```diff
@@ -7,245 +7,95 @@
...
+        "Farsha": {"Cloudsplitter": 0.1},
+        "Tri-edge": {"Ruinator": 0.1},
+        "Lionheart": {"Resolution": 0.1},
+        "Twashtar": {"Rudra's Storm": 0.1},
+        "Carnwenhan": {"Mordant Rime": 0.495},
+        "Drepanum": {"Spiral Hell": 1.0},
+        "Xoanon": {"Retribution": 0.2},
+        "Khatvanga": {"Shattersoul": 0.1},
+        "Vajra": {"Mandalic Stab": 0.495},
+        "Terpsichore": {"Pyrrhic Kleos": 0.495},
```
Comment (IzaKastra) [inline weapon_bonus.py:72]: Terpsichore is already included on line 45.

**Review by IzaKastra — CHANGES_REQUESTED** [review]
I only see two quick issues that should be fixed before merging.

File: weapon_bonus.py (hunk 2 above)
Comment (CaseyMakes) [reply weapon_bonus.py:31]: Yes, it should. Your original code was if 'Heishi' was in the weapon name. Just going on auto-pilot, I wrote it as if it was == instead. Worth checking any other weapons you short handed as well.

File: weapon_bonus.py (hunk 3 above)
Comment (CaseyMakes) [reply weapon_bonus.py:72]: Original code had Vajra applying to both WS. I don't know if it ever came into effect, I haven't looked to see if a WS was job locked yet. Safe to delete the second Terpsichore line.

File: weapon_bonus.py (hunk 2 above)
Comment (IzaKastra) [reply weapon_bonus.py]: Part of my quick check was just seeing if all weapon name strings in your updated code were present as exact matches in the gear.py file's "Name" keys within the mains+ranged lists. Heishi was the only issue.

I ran a similar check against all weapon skill names in the updated code versus weapon skills selectable within the gui_wsdist.py file and found no issues, so all else should be good.

**Pull request merged**